source/grid/manifold.cc

```diff
@@ -683,30 +711,38 @@ FlatManifold<dim, spacedim>::get_new_points(
       surrounding_points_start = modified_points.data();
     }
 
+  for (unsigned int r = 0; r < weights.size(0); ++r)
+    Assert(std::abs(
+             std::accumulate(&weights(r, 0), &weights(r, 0) + n_points, 0.0) -
+             1.0) < 1e-10,
+           ExcMessage("The weights each point should sum to 1!"));
+
   // Now perform the interpolation
-  for (unsigned int row = 0; row < weights.size(0); ++row)
+  for (unsigned int row = 0; row < weights.size(0);)
     {
-      Assert(
-        std::abs(
-          std::accumulate(&weights(row, 0), &weights(row, 0) + n_points, 0.0) -
-          1.0) < 1e-10,
-        ExcMessage("The weights for the individual points should sum to 1!"));
-      Point<spacedim> new_point;
-      for (unsigned int p = 0; p < n_points; ++p)
-        new_point += surrounding_points_start[p] * weights(row, p);
-
-      // if necessary, also adjust the weighted point by the periodicity
-      for (unsigned int d = 0; d < spacedim; ++d)
-        if (periodicity[d] > 0)
-          if (new_point[d] < 0)
-            new_point[d] += periodicity[d];
-
-      // TODO should this use surrounding_points_start or surrounding_points?
-      // The older version used surrounding_points
-      new_points[row] =
-        project_to_manifold(make_array_view(surrounding_points.begin(),
-                                            surrounding_points.end()),
-                            new_point);
+      // For performance reasons, try to compute three points at once
+      if (row + 3 <= weights.size(0))
```

Comment:
Could you make 3 a named variable and use it below. If we should want to change the value in the future, we only need to touch one loc.